### ISSUE TITLE
Add examples in the docs

### ISSUE THREAD

**Issue by svilupp** [JuliaML/OpenAI.jl#54]
It would be excellent to add some examples in the docs.  

I suspect it is hard for newcomers to use the various endpoints and also debug the common error. There is a lot of knowledge how we use these endpoints that is not captured anywhere yet.

Ideas:
- How to call chat endpoint with the new modes (function call/ tools call, json mode, logit bias classifier)
- HTTP debugging and tips (errors when calling wrong model-endpoint combination, dropped calls, timeouts, async challenges (Pool vs Threads))
- Common problems (truncated output for vision models, etc)
- Some practical example to build easy templates with structs?

WDYT? We could probably start small.

**Comment by rory-linehan** [comment]
Agreed - I'm going to start drafting something up while adding the new features.

**Comment by roryl23** [comment]
> did this ever come out?

We have examples in the tests and maybe something in the docs, but I haven't actioned anything with this specifically
